### ISSUE TITLE
Is it possible to allow warnings from one analysis tool, but not others

### ISSUE THREAD

**Issue by stealthrabbi** [novoda/gradle-static-analysis-plugin#226]
e.g. allow warnings in Android Linter, but not SpotBugs

**Comment by tasomaniac** [comment]
Hi, you can do this by 2 possible ways

- some tools allow converting all warnings to errors. Not sure if SpotBugs has that. Android lint has it. 
- you can configure a custom evaluator. You can find some documentation here https://github.com/novoda/gradle-static-analysis-plugin/blob/master/docs/incubating/custom-evaluator.md

**Comment by stealthrabbi** [comment]
OK thanks. And soryr if "issues" are not the right wayt o ask this, but I saw no other way.

So I (temporarily) want to cause builds to fail if there are any warnings or errors in PMD, spotbugs, or checkstyles. And I want it to fail if there are any Errors in Android Lint (warnings are OK).

So perhas a custom validator like so? I assume that if I fail to configure the evaluator correctly, the gradle task itself will fail to run.

```
evaluator { Set allViolations ->
    allViolations.each { violation ->
        if (violation.errors > 0) {
            throw new GradleException("Static Analysis Errors have occurred")
        } else if (violation.name != "Lint"&& violations.warnings > 0) {
            throw new GradleException("non-Lint Static Analysis warnings have occurred")
        }
    }
}
```

**Comment by tasomaniac** [comment]
No problem at all. Let us know if this works. It looks correct to me. The name for Android Lint is just `Lint`. 

The violation class has paths to XML and HTML report files. The original evaluator outputs those for the failed tasks. You might want to keep that.
